src/Datadog.Trace/Agent/Api.cs

```diff
@@ -36,7 +40,19 @@ static Api()
 
         public Api(Uri baseEndpoint, DelegatingHandler delegatingHandler, IStatsd statsd)
         {
-            _tracesEndpoint = new Uri(baseEndpoint, TracesPath);
+            lock (EndpointGate)
+            {
+                _baseEndpoint = baseEndpoint;
+                if (_tracePortOverride != null)
```

Comment:
nit: you might be able to make this more concise by always setting `_tracesEndpoint` to the value returned by the helper URI function and doing some default/null-coalescing logic in there to handle the `_tracePortOverride` setting